test/test_utils/__init__.py

```diff
@@ -321,3 +322,35 @@ def get_dlc_images():
         if dlc_test_type == "sanity":
             return " ".join(images)
     raise RuntimeError(f"Cannot find any images for in {test_images}")
+
+
+def setup_sm_benchmark_tf_train_env(resources_location, setup_tf1_env, setup_tf2_env):
+    """
+    Create a virtual environment for benchmark tests if it doesn't already exist, and download all necessary scripts
+    :param resources_location: <str> directory in which test resources should be placed
+    :param setup_tf1_env: <bool> True if tf1 resources need to be setup
+    :param setup_tf2_env: <bool> True if tf2 resources need to be setup
+    :return: absolute path to the location of the virtual environment
+    """
```

Comment:
This function needs to execute in order to perform a one-time setup before TF SM benchmark tests can be run.